Tribler/community/market/conversion.py

```diff
@@ -0,0 +1,233 @@
+        self.define_meta_message(chr(4), community.get_meta_message(u"offer-sync"),
+                                 self._encode_offer_sync, self._decode_offer_sync)
+        self.define_meta_message(chr(5), community.get_meta_message(u"proposed-trade"),
+                                 self._encode_proposed_trade, self._decode_proposed_trade)
+        self.define_meta_message(chr(6), community.get_meta_message(u"declined-trade"),
+                                 self._encode_declined_trade, self._decode_declined_trade)
+        self.define_meta_message(chr(7), community.get_meta_message(u"counter-trade"),
+                                 self._encode_proposed_trade, self._decode_proposed_trade)
+        self.define_meta_message(chr(8), community.get_meta_message(u"start-transaction"),
+                                 self._encode_start_transaction, self._decode_start_transaction)
+        self.define_meta_message(chr(9), community.get_meta_message(u"wallet-info"),
+                                 self._encode_wallet_info, self._decode_wallet_info)
+        self.define_meta_message(chr(10), community.get_meta_message(u"payment"),
+                                 self._encode_payment, self._decode_payment)
+        self.define_meta_message(chr(11), community.get_meta_message(u"end-transaction"),
+                                 self._encode_transaction, self._decode_transaction)
+
+    def _encode_introduction_request(self, message):
+        data = BinaryConversion._encode_introduction_request(self, message)
+
+        if message.payload.orders_bloom_filter:
+            data.extend((pack('!BH', message.payload.orders_bloom_filter.functions,
+                              message.payload.orders_bloom_filter.size), message.payload.orders_bloom_filter.prefix,
+                         message.payload.orders_bloom_filter.bytes))
+        return data
+
+    def _decode_introduction_request(self, placeholder, offset, data):
+        offset, payload = BinaryConversion._decode_introduction_request(self, placeholder, offset, data)
+
+        if len(data) > offset:
+            if len(data) < offset + 5:
+                raise DropPacket("Insufficient packet size")
+
+            functions, size = unpack_from('!BH', data, offset)
+            offset += 3
+
+            prefix = data[offset]
+            offset += 1
+
+            if functions <= 0 or size <= 0 or size % 8 != 0:
+                raise DropPacket("Invalid bloom filter")
+
+            length = int(ceil(size / 8))
+            if length != len(data) - offset:
+                raise DropPacket("Invalid number of bytes available (irq) %d, %d, %d" %
+                                 (length, len(data) - offset, size))
+
+            orders_bloom_filter = BloomFilter(data[offset:offset + length], functions, prefix=prefix)
+            offset += length
+
+            payload.set_orders_bloom_filter(orders_bloom_filter)
+
```

Comment:
The bloom filter in an introduction request is optional. If I would raise a `DropPacket` here, we would be unable to talk to the bootstrap servers.